### ISSUE TITLE
Invalid label crashes analyzer with NNBD enabled

### ISSUE THREAD

**Issue by munificent** [dart-lang/sdk#42060]
Take:

```dart
main() {
  var L = 33;
  while (false) if (true) break L;
}
```

Run:

```sh
$ dartanalyzer --enable-experiment=non-nullable temp.dart
```

Behold:

```
Unhandled exception:
type 'LocalVariableElementImpl' is not a subtype of type 'LabelElement'
#0      FlowAnalysisHelper.breakStatement (package:analyzer/src/dart/resolver/flow_analysis_visitor.dart:111:44)
#1      ResolverVisitor.visitBreakStatement (package:analyzer/src/generated/resolver.dart:677:20)
#2      BreakStatementImpl.accept (package:analyzer/src/dart/ast/ast.dart:1176:49)
#3      ScopedVisitor.visitStatementInScope (package:analyzer/src/generated/resolver.dart:2635:14)
#4      ResolverVisitor.visitIfStatement (package:analyzer/src/generated/resolver.dart:1194:7)
#5      IfStatementImpl.accept (package:analyzer/src/dart/ast/ast.dart:5670:49)
#6      ScopedVisitor.visitStatementInScope (package:analyzer/src/generated/resolver.dart:2635:14)
#7      ResolverVisitor.visitWhileStatement (package:analyzer/src/generated/resolver.dart:1696:9)
#8      WhileStatementImpl.accept (package:analyzer/src/dart/ast/ast.dart:10721:49)
#9      NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:7586:20)
#10     BlockImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:1083:17)
#11     ResolverVisitor.visitNode (package:analyzer/src/generated/resolver.dart:1367:10)
#12     UnifyingAstVisitor.visitBlock (package:analyzer/dart/ast/visitor.dart:3128:31)
#13     ScopedVisitor.visitBlock (package:analyzer/src/generated/resolver.dart:2075:13)
#14     BlockImpl.accept (package:analyzer/src/dart/ast/ast.dart:1079:49)
#15     BlockFunctionBodyImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:1038:13)
#16     ResolverVisitor.visitNode (package:analyzer/src/generated/resolver.dart:1367:10)
#17     UnifyingAstVisitor.visitBlockFunctionBody (package:analyzer/dart/ast/visitor.dart:3131:55)
#18     ScopedVisitor.visitBlockFunctionBody (package:analyzer/src/generated/resolver.dart:2086:13)
#19     ResolverVisitor.visitBlockFunctionBody (package:analyzer/src/generated/resolver.dart:657:13)
#20     BlockFunctionBodyImpl.accept (package:analyzer/src/dart/ast/ast.dart:1034:49)
#21     FunctionExpressionImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:5006:12)
#22     FunctionExpressionResolver.resolve (package:analyzer/src/dart/resolver/function_expression_resolver.dart:64:10)
#23     ResolverVisitor.visitFunctionExpression (package:analyzer/src/generated/resolver.dart:1091:35)
#24     FunctionExpressionImpl.accept (package:analyzer/src/dart/ast/ast.dart:5000:49)
#25     FunctionDeclarationImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:4863:26)
#26     ResolverVisitor.visitNode (package:analyzer/src/generated/resolver.dart:1367:10)
#27     UnifyingAstVisitor.visitFunctionDeclaration (package:analyzer/dart/ast/visitor.dart:3256:59)
#28     ScopedVisitor.visitFunctionDeclarationInScope (package:analyzer/src/generated/resolver.dart:2417:11)
#29     ResolverVisitor.visitFunctionDeclarationInScope (package:analyzer/src/generated/resolver.dart:1081:11)
#30     ScopedVisitor.visitFunctionDeclaration (package:analyzer/src/generated/resolver.dart:2410:7)
#31     ResolverVisitor.visitFunctionDeclaration (package:analyzer/src/generated/resolver.dart:1050:11)
#32     FunctionDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:4856:49)
#33     ResolverVisitor.visitCompilationUnit (package:analyzer/src/generated/resolver.dart:754:23)
#34     CompilationUnitImpl.accept (package:analyzer/src/dart/ast/ast.dart:2125:49)
#35     LibraryAnalyzer._resolveFile (package:analyzer/src/dart/analysis/library_analyzer.dart:733:10)
#36     LibraryAnalyzer.analyzeSync.<anonymous closure> (package:analyzer/src/dart/analysis/library_analyzer.dart:148:7)
#37     _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:377:8)
#38     LibraryAnalyzer.analyzeSync (package:analyzer/src/dart/analysis/library_analyzer.dart:147:11)
#39     LibraryAnalyzer.analyze.<anonymous closure> (package:analyzer/src/dart/analysis/library_analyzer.dart:116:14)
#40     _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:256:15)
#41     LibraryAnalyzer.analyze (package:analyzer/src/dart/analysis/library_analyzer.dart:115:43)
#42     AnalysisDriver._computeAnalysisResult2.<anonymous closure> (package:analyzer/src/dart/analysis/driver.dart:1313:63)
#43     PerformanceLog.run (package:analyzer/src/dart/analysis/performance_logger.dart:34:15)
#44     AnalysisDriver._computeAnalysisResult2 (package:analyzer/src/dart/analysis/driver.dart:1289:20)
#45     AnalysisDriver._computeAnalysisResult.<anonymous closure> (package:analyzer/src/dart/analysis/driver.dart:1240:14)
#46     _rootRun (dart:async/zone.dart:1190:13)
#47     _CustomZone.run (dart:async/zone.dart:1093:19)
#48     _runZoned (dart:async/zone.dart:1630:10)
#49     runZoned (dart:async/zone.dart:1550:10)
#50     NullSafetyUnderstandingFlag.enableNullSafetyTypes (package:analyzer/dart/element/null_safety_understanding_flag.dart:42:12)
#51     AnalysisDriver._computeAnalysisResult (package:analyzer/src/dart/analysis/driver.dart:1239:40)
#52     AnalysisDriver.getErrors (package:analyzer/src/dart/analysis/driver.dart:565:41)
#53     AnalyzerImpl.prepareErrors (package:analyzer_cli/src/analyzer_impl.dart:129:49)
<asynchronous suspension>
#54     AnalyzerImpl._analyze (package:analyzer_cli/src/analyzer_impl.dart:168:11)
<asynchronous suspension>
#55     AnalyzerImpl.analyze (package:analyzer_cli/src/analyzer_impl.dart:105:18)
<asynchronous suspension>
#56     Driver._runAnalyzer (package:analyzer_cli/src/driver.dart:636:21)
#57     Driver._analyzeAllImpl (package:analyzer_cli/src/driver.dart:353:30)
#58     _RootZone.runUnary (dart:async/zone.dart:1450:54)
#59     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
#60     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
#61     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
#62     Future._addListener.<anonymous closure> (dart:async/future_impl.dart:393:9)
#63     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#64     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#65     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
#66     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)

#0      AnalysisDriver._computeAnalysisResult2.<anonymous closure> (package:analyzer/src/dart/analysis/driver.dart:1349:9)
#1      PerformanceLog.run (package:analyzer/src/dart/analysis/performance_logger.dart:34:15)
#2      AnalysisDriver._computeAnalysisResult2 (package:analyzer/src/dart/analysis/driver.dart:1289:20)
#3      AnalysisDriver._computeAnalysisResult.<anonymous closure> (package:analyzer/src/dart/analysis/driver.dart:1240:14)
#4      _rootRun (dart:async/zone.dart:1190:13)
#5      _CustomZone.run (dart:async/zone.dart:1093:19)
#6      _runZoned (dart:async/zone.dart:1630:10)
#7      runZoned (dart:async/zone.dart:1550:10)
#8      NullSafetyUnderstandingFlag.enableNullSafetyTypes (package:analyzer/dart/element/null_safety_understanding_flag.dart:42:12)
#9      AnalysisDriver._computeAnalysisResult (package:analyzer/src/dart/analysis/driver.dart:1239:40)
#10     AnalysisDriver.getErrors (package:analyzer/src/dart/analysis/driver.dart:565:41)
#11     AnalyzerImpl.prepareErrors (package:analyzer_cli/src/analyzer_impl.dart:129:49)
<asynchronous suspension>
#12     AnalyzerImpl._analyze (package:analyzer_cli/src/analyzer_impl.dart:168:11)
<asynchronous suspension>
#13     AnalyzerImpl.analyze (package:analyzer_cli/src/analyzer_impl.dart:105:18)
<asynchronous suspension>
#14     Driver._runAnalyzer (package:analyzer_cli/src/driver.dart:636:21)
#15     Driver._analyzeAllImpl (package:analyzer_cli/src/driver.dart:353:30)
#16     _RootZone.runUnary (dart:async/zone.dart:1450:54)
#17     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
#18     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
#19     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
#20     Future._addListener.<anonymous closure> (dart:async/future_impl.dart:393:9)
#21     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#22     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#23     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
#24     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)
```

If you want a corresponding language test, `tests/language/label/label5_test.dart` has the same problem.

**Comment by munificent** [comment]
cc @stereotype441

**Comment by scheglov** [comment]
Fixed in 860d3c0697e2bff140a0d956a8d1ccd46348d13e as https://github.com/dart-lang/sdk/issues/42077.
